### ISSUE TITLE
global env overrides local with

### ISSUE THREAD

**Issue by marekdedic** [shivammathur/setup-php#295]
**Describe the bug**
Hi, I had the following code in my action file:

```yml
name: "CI"
on:
  push:
    branches: "*"
  pull_request:
env:
  php-version: 7.4
  cache-version: 1
jobs:
  platform-check:
    name: "PHP 5.6 compatibility"
    runs-on: ubuntu-latest
    steps:
      - name: "Checkout"
        uses: actions/checkout@v2

      - name: "Setup PHP"
        uses: shivammathur/setup-php@v2
        with:
          php-version: 5.6
          coverage: none

      - name: "Check dependency compatibility"
        run: |
          composer check-platform-reqs --no-dev
```

The `env` variable was then used for all the jobs **except** the one described in the snippet. However, with this config, the global `env` overrode the local `with` and I got version 7.4 for the job! Now, I agree that it was probably stupid of me to choose the same name, however, I would really expect this to use 5.6 for this job...

**Version**
- [x] I have checked releases and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**
ubuntu

**Additional context**
See skaut/skaut-google-drive-gallery#671 for full run/patch.

**Are you willing to submit a PR?**
No, I have no idea how to fix this.

**Comment by shivammathur** [comment]
@marekdedic This should be fixed now after `2.6.0` release.  
`with` will be preferred over `env`.

**Comment by marekdedic** [comment]
Hi,
thanks for the fix, it works :)
